bitbots_vision/scripts/evaluator.py

```diff
@@ -0,0 +1,738 @@
+            return
+
+
+        # Setting label size as used by the imagetagger before we do the resizeing
+        if self._label_shape is None:
+            self._label_shape = image.shape[:-1]
+
+
+        # Set the resized image size as expected by the vision
+        if self._image_shape is None:
+            self._image_shape = np.array((
+                rospy.get_param("~image_resolution_y"), 
+                rospy.get_param("~image_resolution_x")))
+
+        # resize the image
+        image = cv2.resize(image, tuple(np.flip(self._image_shape)), interpolation=3)
+
+        # Building and sending message
+        rospy.loginfo('Sending image {} of {} (starting by 0).'.format(self._current_image_counter, self._image_count))
+        msg = self.bridge.cv2_to_imgmsg(image, 'bgr8')
+        msg.header.stamp = rospy.Time.now()
+        # The second unforgivable curse. 
+        # The tf frame name is abused to store the id of the image
+        msg.header.frame_id = str(self._current_image_counter)
+        self._image_pub.publish(msg)
+
+        # Set up evaluation element in measurements list
+        self._measurements[self._current_image_counter] = ImageMeasurement(self._images[self._current_image_counter], self._evaluated_classes)
+
+    def _read_labels(self, filename):
+        """
+        Reads the labels YAML file and returns a list of image names with their labels.
+        This is set up to work with the "evaluation" export format of the Bit-Bots Team in the ImageTagger.
+        """
+        filepath = os.path.join(self._image_path, filename)
+        images = None
+        if not os.path.isfile(filepath):
+            rospy.logerr('File at path \'{}\' not found!'.format(filepath))
+            return
+        pickle_filepath = filepath + '.pickle'
+        if not os.path.isfile(pickle_filepath):
+            rospy.loginfo('Creating pickle file from yaml...')
+            with open(filepath, 'r') as stream:
+                try:
+                    images = yaml.load(stream)['labels']
+                except yaml.YAMLError as exc:
+                    rospy.logerr(exc)
+            with open(pickle_filepath, 'wb') as f:
+                pickle.dump(images, f)
+            rospy.loginfo('Done.')
+        else:
+            rospy.loginfo('Reading pickle file...')
+            try:
+                with open(pickle_filepath, 'rb') as f:
+                    images = pickle.load(f)
+            except pickle.PickleError as exc:
+                rospy.logerr(exc)
+        return images
+
+    def _get_image_measurement(self, image_id):
+        """
+        Reads the mesurement env for a specific image
+        """
+        if image_id not in self._measurements.keys():
+            rospy.logerr('got an unknown image with seq {}! Is there a ROS-bag running? Stop it please!'.format(image_sequence))
+            return
+        return self._measurements[image_id]
+
+    def _balls_callback(self, msg):
```

Comment:
All of these callbacks have a lot of duplicated code... This should be refactored or at least create an issue